### PR TITLE
#16298 this sould fix the issue and fix a test fails on previous PR

### PR DESCRIPTION
**Pull request by sichensol**
### Fixed Issues (if relevant)

1. magento/magento2#16298: Magento sometimes doesn't select the correct store if the scope code is null
2. Fixed PR https://github.com/fratob85/magento2/tree/fix-switch-to-incorrect-store

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @molneek. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-3011](https://jira.corp.magento.com/browse/ENGCOM-3011) has been created to process this Pull Request

File: lib/internal/Magento/Framework/App/Config/ScopeCodeResolver.php [hunk 1]
```diff
@@ -34,10 +34,10 @@ public function __construct(ScopeResolverPool $scopeResolverPool)
      * Resolve scope code
      *
      * @param string $scopeType
-     * @param string $scopeCode
+     * @param null|string $scopeCode
      * @return string
      */
-    public function resolve($scopeType, $scopeCode)
+    public function resolve($scopeType, $scopeCode = null)
```
Comment (sidolov) [inline lib/internal/Magento/Framework/App/Config/ScopeCodeResolver.php:40]: We are not allowed to modify public methods signature according to our [Backward compatible development guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @molneek , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**